build.cake

```diff
@@ -0,0 +1,304 @@
+            }
+
+            Information("");
+        }
+    }
+    
+    if (!success)
+    {
+        throw new CakeException("There was an error while executing the tests");
+    }
+    
+    if (!success)
+    {
+        throw new CakeException("There was an error while executing the tests");
+    }
+
+    string[] GetTargetFrameworks(FilePath file)
+    {
+        XmlPeekSettings settings = new XmlPeekSettings
+        {
+            SuppressWarning = true
+        };
+
+        return (XmlPeek(file, "/Project/PropertyGroup/TargetFrameworks", settings) ?? XmlPeek(file, "/Project/PropertyGroup/TargetFramework", settings)).Split(";");
+    }
+});
+
+Task("MergeCoverageResults")
+    .IsDependentOn("RunTests")
+    .Does<BuildState>(state =>
+{
+    Information("Merging coverage files");
+    var coverageFiles = GetFiles($"{state.Paths.TestOutputFolder}/*.dcvr");
+    DotCoverMerge(coverageFiles, state.Paths.DotCoverOutputFile);
+    DeleteFiles(coverageFiles);
+});
+
+Task("GenerateXmlReport")
+    .IsDependentOn("MergeCoverageResults")
+    .Does<BuildState>(state =>
+{
+    Information("Generating dotCover XML report");
+    DotCoverReport(state.Paths.DotCoverOutputFile, state.Paths.DotCoverOutputFileXml, new DotCoverReportSettings 
+    {
+        ReportType = DotCoverReportType.DetailedXML
+    });
+});
+
+Task("ExportReport")
+    .IsDependentOn("GenerateXmlReport")
+    .Does<BuildState>(state =>
+{
+    Information("Executing ReportGenerator to generate HTML report");
+    ReportGenerator(state.Paths.DotCoverOutputFileXml, state.Paths.ReportFolder, new ReportGeneratorSettings {
+            ReportTypes = new[]{ReportGeneratorReportType.Html, ReportGeneratorReportType.Xml}
+    });
+});
+
+Task("UploadTestsToAppVeyor")
+    .IsDependentOn("RunTests")
+    .WithCriteria(BuildSystem.IsRunningOnAppVeyor)
+    .Does<BuildState>(state =>
+{
+    Information("Uploading test result files to AppVeyor");
+    var testResultFiles = GetFiles($"{state.Paths.TestOutputFolder}/*.trx");
+
+    foreach (var file in testResultFiles)
+    {
+        Information($"\tUploading {file.GetFilename()}");
+        try
+        {
+            AppVeyor.UploadTestResults(file, AppVeyorTestResultsType.MSTest);
+        }
+        catch 
+        {
+            Error($"Failed to upload {file.GetFilename()}");
+        }
+    }
+});
+
+Task("Test")
+    .IsDependentOn("RunTests")
+    .IsDependentOn("MergeCoverageResults")
+    .IsDependentOn("GenerateXmlReport")
+    .IsDependentOn("ExportReport")
+    .IsDependentOn("UploadTestsToAppVeyor");
+
+Task("PackTool")
+    .IsDependentOn("Version")
+    .IsDependentOn("Restore")
+    .Does<BuildState>(state =>
+{
+    var settings = new DotNetCorePackSettings
+    {
+        Configuration = "Release",
+        NoRestore = true,
+        OutputDirectory = state.Paths.OutputFolder,
+        MSBuildSettings = new DotNetCoreMSBuildSettings()
+                            .SetInformationalVersion(state.Version.AssemblyVersion)
```

Comment:
Used to support version output from the tool itself using the `--version` option